bundles/org.openhab.binding.luftdateninfo/src/main/java/org/openhab/binding/luftdateninfo/internal/handler/ConditionHandler.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luftdateninfo.internal.handler;
+
+import static org.openhab.binding.luftdateninfo.internal.LuftdatenInfoBindingConstants.*;
+import static org.openhab.binding.luftdateninfo.internal.handler.HTTPHandler.*;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.openhab.binding.luftdateninfo.internal.dto.SensorDataValue;
+import org.openhab.binding.luftdateninfo.internal.utils.NumberUtils;
+
+/**
+ * The {@link ConditionHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ConditionHandler extends BaseSensorHandler {
+
+    protected DecimalType temperatureCache = UNDEF;
+    protected DecimalType humidityCache = UNDEF;
+    protected DecimalType pressureCache = UNDEF;
+    protected DecimalType pressureSeaCache = UNDEF;
+
+    public ConditionHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public int updateChannels(@Nullable String json) {
+        if (json != null) {
+            List<SensorDataValue> valueList = HTTPHandler.getLatestValues(json);
+            if (valueList != null) {
+                if (HTTPHandler.isCondition(valueList)) {
+                    Iterator<SensorDataValue> iter = valueList.iterator();
+                    while (iter.hasNext()) {
+                        SensorDataValue v = iter.next();
+                        if (v.getValue_type().equals(TEMPERATURE)) {
+                            temperatureCache = new DecimalType(NumberUtils.round(v.getValue(), 1));
```

Comment:
When using Units of Measure, this would look like:
```suggestion
                            temperatureCache = QuantityType.valueOf(NumberUtils.round(v.getValue(), 1),
                                    SIUnits.CELSIUS);
```

Reply:
All values adapted to QuantityType